### ISSUE TITLE
Show a better error on mixing list and key-value syntax in the rules entry.  

### ISSUE THREAD

**Issue by dark-chocolate** [dart-lang/sdk#43320]
Put this code in your `analysis_options.yaml` file: 

```
linter:
  rules:
    close_sinks: false
    - empty_statements 
```
The error says:

> Expected a key while parsing a block mapping.

I think this error isn't quite clear. The docs show a clearer error though, quoting [this](https://dart.dev/guides/language/analysis-options#disabling-individual-rules): 

> You can’t mix list and key-value syntax in the same rules entry. 

Can Dart team change the error to make it more clearer or is me who didn't get the error at first before reading the actual line mentioned in the documentation

**Comment by bwilkerson** [comment]
I agree that the message isn't very clear. That message is currently coming from `package:yaml` as a parse error, and the analyzer is just surfacing the message. I can think of two possible fixes:

1. We could try to get `package:yaml` to output a better message. I've never looked at the package so I don't know how hard it would be, but that would be my preference because that would benefit all users of the package.

2. We could look for this message in the analyzer and translate it before displaying it. This solution would be brittle (if `package:yaml` ever changed the message our code would break) and would only help users of the analyzer, so it's less attractive to me.

I created an [issue](https://github.com/dart-lang/yaml/issues/88) for the package to see whether that's a viable path forward.
